ldk/javascript/examples/self-test-loop/src/tests/index.ts

```diff
@@ -921,6 +922,75 @@ export const networkHTTP = (): Promise<boolean> =>
       });
   });
 
+export const networkWebSocket = (): Promise<boolean> =>
+  new Promise(async (resolve, reject) => {
+    const url = 'wss://html5rocks.websocket.org/echo';
+    const testData = new Uint8Array([53, 6, 6, 65, 20, 74, 65, 78, 74]);
+    const testText = 'some text';
+    let textTestPassed = false;
+    let binaryTestPassed = false;
+
+    setTimeout(() => {
+      reject(new Error('Network websocket test did not finish in the appropriate timespan.'));
+    }, 20000);
+
+    const socketConfiguration: network.SocketConfiguration = {
+      url: url,
+      useCompression: true,
+    };
+
+    try {
+      console.debug('Stating websocket connect');
+      const socket = await network.webSocketConnect(socketConfiguration);
+      console.debug('Websocket successfully connected');
+      socket.setCloseHandler((error, code, text) => {
+        if (error) {
+          console.error(`OnCloseHandler received error:`);
+          console.error(error);
+          
+          return;
+        }
+
+        console.info(`Received on close code: ${code}. ${text}`);
+      });
+      const cancellable: Cancellable = await socket.setMessageHandler(async (error, message) => {
+        if (error) {
+          console.error(error);
+
+          return;
+        }
+        if (message) {
+          if (typeof message === 'string') {
+            if (message === testText) {
+              console.debug(`Received text data`);
+              textTestPassed = true;
+              if (binaryTestPassed) {
+                resolve(true);
+                await testUtils.finalizeWebsocketTest(cancellable, socket);
+              }
+            }
+          } else {
+            if (JSON.stringify(message) === JSON.stringify(testData)) {
+              console.debug(`Received binary data`);
+              binaryTestPassed = true;
+              if (textTestPassed) {
```

Comment:
What is the purpose of having this and `binaryTestPassed`? Does the connection you're using work in some way where it sends both messages?